packages/delisp-core/src/sexpr.ts

```diff
@@ -22,6 +22,7 @@ export interface SExprArray extends Array<ASExpr> {}
 
 export interface ASExprList {
   type: "list";
+  shape: "round" | "square";
```

Comment:
Seems all correct, but a bit non-lispy.  Instead of an attribute to the lists, and make them round/square, what other lisps (emacs lisp & clojure) do is to just call `[1 2 3 4]` vector or array.

That would mean defining a new type of S-Expression.  But I think it makes it a bit more clear.

Reply:
Also it is more backward-compatible, you wouldn't need to change any tests, and safer,  as if we forget to check the `shape` attribute, we may accept square backets (vector) where a list is expected.